docusaurus/docs/iOS/guides/working-with-the-composer.md

```diff
@@ -0,0 +1,221 @@
+
+## Customizing the Composer UI
+By subclassing the `ComposerView` you can customize the layout of the Composer, for example, changing the position of the views, removing views or adding new ones. Since all the logic of the composer lives on the `ComposerVC`, this is the class to override in case the customization depends on data. Here is an example of customizing the composer to look like the iMessage app:
+```swift
+final class iMessageComposerView: ComposerView {
+    override func setUpLayout() {
+        super.setUpLayout()
+
+        leadingContainer.removeArrangedSubview(commandsButton)
+        trailingContainer.removeArrangedSubview(sendButton)
+        inputMessageView.inputTextContainer.addArrangedSubview(sendButton)
+    }
+    
+    override func setUpAppearance() {
+        super.setUpAppearance()
+
+        attachmentButton.setImage(
+            UIImage(systemName: "camera.fill"),
+            for: .normal
+        )
+    }
+}
+
+final class iMessageComposerVC: ComposerVC {
+    override func updateContent() {
+        super.updateContent()
+
+        switch content.state {
+        case .new:
+            composerView.inputMessageView.textView.placeholderLabel.text = "iMessage"
+        default:
+            break
+        }
+    }
+}
+
+// Then, replace the components:
+components.messageComposerVC = iMessageComposerVC.self
+components.messageComposerView = iMessageComposerView.self
+```
+
+**Result:**
+<img src={require("../assets/composer-ui-custom-layout.png").default} width="50%"/>
+
+Since the Composer is built with multiple `ContainerStackView`'s with a similar API of a `UIStackView`, it is very easy to change the Composer layout it is just a question of removing views from containers and adding them back to the desired containers. In this case, we override the `setUpLayout()` lifecycle to remove the `commandsButton` all together and we move the `sendButton` from the `trailingContainer` to the `inputTextContainer`. As a last touch, we change the `attachmentButton` image by overriding the `setUpAppearance()` lifecycle.
+
+## Customizing the Composer Actions
+All the actions triggered by the Composer buttons are wired in the `ComposerVC.setUp()` lifecycle. When adding new buttons it is possible to add new actions by overriding this function. Let's imagine an example where we have added a new custom emoji button to open an emoji picker:
+```swift
+final class CustomComposerVC: ComposerVC {
+    override func setUp() {
+        super.setUp()
+
+        composerView.emojiButton.addTarget(self, action: #selector(showEmojiPicker), for: .touchUpInside)
```

Comment:
Let's show introducing this button to the layout, too

Reply:
You mean adding the emoji button to the ComposerView layout right? Yes makes sense 👍